framework/source/class/qx/ui/window/Window.js

```diff
@@ -1050,5 +1210,29 @@ qx.Class.define("qx.ui.window.Window",
       this.close();
       this.getChildControl("close-button").reset();
     }
+  },
+
+  destruct : function()
+  {
+    var id;
+    var parent;
+
+    // Remove ourselves from the focus handler
+    qx.ui.core.FocusHandler.getInstance().removeRoot(this);
+
+    // Remove the listener for appear, if there is one
+    id = this.__centeringAppearId;
+    id && this.removeListenerById(id);
```

Comment:
Yes, I need to track the listener id so that I can remove the listener if the property value changes. (It's much faster to use the listener id than to search through the list of listeners for the one that matches a given function and context.) But yes, I can remove these two lines here because all listeners of `this` are removed in higher-layer code.